qiskit/circuit/library/standard_gates/equivalence_library.py

```diff
@@ -850,6 +850,77 @@ def _cnot_rxx_decompose(plus_ry: bool = True, plus_rxx: bool = True):
     def_swap.append(inst, qargs, cargs)
 _sel.add_equivalence(SwapGate(), def_swap)
 
+# SwapGate
+#
+# q_0: ─X─
+#       │   ≡
+# q_1: ─X─
+#
+# global phase: π/2
+#      ┌─────────┐┌────┐          ┌──────┐┌──────────┐┌────┐┌────────┐┌──────┐»
+# q_0: ┤ Rz(π/2) ├┤ √X ├──────────┤0     ├┤ Rz(-π/2) ├┤ √X ├┤ Rz(-π) ├┤0     ├»
+#      ├─────────┤├────┤┌────────┐│  Ecr │├─────────┬┘├────┤└────────┘│  Ecr │»
+# q_1: ┤ Rz(π/2) ├┤ √X ├┤ Rz(-π) ├┤1     ├┤ Rz(π/2) ├─┤ √X ├──────────┤1     ├»
+#      └─────────┘└────┘└────────┘└──────┘└─────────┘ └────┘          └──────┘»
+# «     ┌─────────┐ ┌────┐┌─────────┐┌──────┐┌──────────┐┌────┐┌──────────┐
+# «q_0: ┤ Rz(π/2) ├─┤ √X ├┤ Rz(π/2) ├┤0     ├┤ Rz(-π/2) ├┤ √X ├┤ Rz(-π/2) ├
+# «     ├─────────┴┐└────┘└─────────┘│  Ecr │├─────────┬┘├────┤├──────────┤
+# «q_1: ┤ Rz(-π/2) ├─────────────────┤1     ├┤ Rz(π/2) ├─┤ √X ├┤ Rz(-π/2) ├
+# «     └──────────┘                 └──────┘└─────────┘ └────┘└──────────┘
+#
+q = QuantumRegister(2, "q")
+def_swap_ecr = QuantumCircuit(q)
+def_swap_ecr.rz(-pi / 2, 0)
+def_swap_ecr.sx(1)
+def_swap_ecr.ecr(0, 1)
+def_swap_ecr.rz(-pi / 2, 1)
+def_swap_ecr.sx(0)
+def_swap_ecr.ecr(1, 0)
+def_swap_ecr.rz(-pi / 2, 0)
+def_swap_ecr.sx(1)
+def_swap_ecr.ecr(0, 1)
+_sel.add_equivalence(SwapGate(), def_swap_ecr)
+
```

Comment:
There are 23 gates in the diagram and only 9 gates in the code. Is this correct?

Reply:
Oops good catch, yeah I updated the circuit based on @alexanderivrii's more efficient circuit from: https://github.com/Qiskit/qiskit/pull/12312#issuecomment-2088785155 but neglected to update the code comment. Let me do that now.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12312/commits/1354c2f1593c691eb913caf8908600167ab8b3bc